### PR TITLE
Centralize cache invalidation on file urls

### PR DESCRIPTION
**Pull request by Speedphoenix**

### CODE REVIEW

File: hide/comp/Scene.hx [hunk 1]
```diff
@@ -258,7 +258,7 @@ class Scene extends Component implements h3d.IDrawable {
 
 	function _loadTextureData( img : hxd.res.Image, onReady : Void -> Void, t : h3d.mat.Texture ) {
 		var path = ide.getPath(img.entry.path);
-		var img = new Element('<img src="file://$path" crossorigin="anonymous"/>');
+		var img = new Element('<img src="file://${ide.getUnCachedUrl(path)}" crossorigin="anonymous"/>');
```
Comment (trethaller) [inline hide/comp/Scene.hx:261]: looks like you'll get `file://` twice here?

File: hide/Ide.hx [hunk 2]
```diff
@@ -823,6 +823,11 @@ class Ide {
 		return path;
 	}
 
+	public function getUnCachedUrl( path : String ) {
```
Comment (trethaller) [inline hide/Ide.hx:826]: Could be a static

**Review by trethaller — CHANGES_REQUESTED** [review]

File: hide/Ide.hx (hunk 2 above)
Comment (Speedphoenix) [reply hide/Ide.hx:826]: But it uses `getPath()` which is not a static function (uses the `resourceDir` class variable).

Shall I make it static and call `inst.getPath()`?

File: hide/comp/Scene.hx (hunk 1 above)
Comment (Speedphoenix) [reply hide/comp/Scene.hx]: Yes, thank you.

Fixed now on 6d061c6

**Pull request merged**